src/main/java/com/netflix/msl/entityauth/X509AuthenticationData.java

```diff
@@ -97,18 +114,15 @@ public X509AuthenticationData(final X509Certificate x509cert) throws MslCryptoEx
         }
         
         // Create X.509 cert.
-        final byte[] x509bytes;
-        try {
-            x509bytes = DatatypeConverter.parseBase64Binary(x509);
-        } catch (final IllegalArgumentException e) {
-            throw new MslCryptoException(MslError.X509CERT_INVALID, x509, e);
-        }
         try {
-            final ByteArrayInputStream bais = new ByteArrayInputStream(x509bytes);
-            x509cert = (X509Certificate)factory.generateCertificate(bais);
-            identity = x509cert.getSubjectX500Principal().getName();
+            final ByteArrayInputStream bais = new ByteArrayInputStream(inBytes);
+            certs = (Collection<X509Certificate>)factory.generateCertificates(bais);
+            if (certs.isEmpty()) {
+            	throw new MslCryptoException(MslError.X509CERT_PARSE_ERROR, inString, null);
```

Comment:
Should factory.generateCertificates() throw an exception instead of returning the empty list? Because empty list might imply the string value is empty. Which could be checked for independently of an invalid cert.